packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -610,17 +557,18 @@ export abstract class BucketBase extends Resource implements IBucket {
    */
   protected abstract disallowPublicAccess?: boolean;
 
-  private notifications?: BucketNotifications;
+  protected factories: BucketFactories;
 
-  protected notificationsHandlerRole?: iam.IRole;
+  public notificationsHandlerRole?: iam.IRole;
```

Comment:
Funny enough: I think this breaks the singleton notifications handler (by ignoring the second bucket with a different handler role).